### PR TITLE
added docs for config 2.0 and search

### PR DESCRIPTION
**Pull request by jainabhinav**

### CODE REVIEW

File: docs/low-code-spark/configuration.md [hunk 1]
```diff
@@ -9,76 +9,103 @@ tags:
...
-   This will be compiled as:
+Prophecy IDE allows you to define three kinds of configurations:
 
-   ```python
-   spark.conf.set("spark.sql.autoBroadcastJoinThreshold", "10485760")
-   ```
+## Spark Configuration
 
-   :::<br/>
+Runtime Spark configurations as name-value pairs. The name-value pairs will be set inside the Spark runtime configurations as `spark.conf.set(name, value)`
```
Comment (slyons) [inline docs/low-code-spark/configuration.md:19]: "Runtime Spark configurations" -> "Set runtime Spark configurations"

File: docs/low-code-spark/configuration.md [hunk 2]
```diff
@@ -9,76 +9,103 @@ tags:
...
----
+![Configurations - Spark](./img/configs_hadoop.png)
+
+## Pipeline Configuration
+
+Config values which can be set at Pipeline level and then be accessed inside any component in the Pipeline. [Multiple instances](#pipeline-configuration-instances)
+of configuration can be created per Pipeline.
+![Configurations - Common](img/config-pipeline-eg1.png)
+
+Syntax for using configuration inside Gems:
```
Comment (slyons) [inline docs/low-code-spark/configuration.md:35]: Can we turn this into a visual example with a Code block? We might have to break this up based on the use case a bit more. The user won't understand what "visual language SQL" vs "visual language Scala" means.

File: docs/low-code-spark/configuration.md [hunk 3]
```diff
@@ -9,76 +9,103 @@ tags:
...
+
+In the below image `'$num_top_customers'` is fetching the integer value defined in configurations.
+
+![Config Limit Example](img/config-pipeline-limit-eg.png)
+
+### Using Spark-expression Config type in Gem
+
+Here we have used Spark expression directly from our config value to populate a column.
+
+In the below image `amounts` -> `expr('$test_expression')` is coming from configuration type defined as `Spark-expression`
```
Comment (slyons) [inline docs/low-code-spark/configuration.md:58]: Add numbers/letters to the image to draw the reader's attention to them, then put a table under the image with the explanation. See the Concepts section of documents for some examples.

File: docs/low-code-spark/configuration.md [hunk 4]
```diff
@@ -9,76 +9,103 @@ tags:
...
+and `report_name` -> `'$report_name'` is coming configuration type defined as string.
+
+![Config Reformat example](img/config-pipeline-reformat-eg.png)
+
+Similarly configurations defined as type Spark-expression can be used directly in filter, join, reformat etc.
+
+### Using config in paths for Source/Target Gems
+
+Config can also be used to refer to paths. This type of configuration comes in handy in situation where we have dev, qa and prod data.
```
Comment (slyons) [inline docs/low-code-spark/configuration.md:67]: "dev" -> "`DEV`", for dev qa and prod

**Review by slyons — CHANGES_REQUESTED** [review]

**Review by rishabh-sudan — APPROVED** [review]

**Pull request merged**